instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/snippet/ServletOutputStreamInjectionHelper.java

```diff
@@ -42,8 +42,16 @@ public static boolean handleWrite(
     state.setAlreadyInjected(); // set before write to avoid recursive loop
     out.write(original, off, i + 1);
     try {
-      byte[] snippetBytes = SnippetHolder.getSnippetBytes(state.getCharacterEncoding());
+      byte[] snippetBytes = ExperimentalSnippetHolder.getSnippetBytes(state.getCharacterEncoding());
+      if (state.getWrapper().isCommitted()) {
+        // header already set and sent, don't inject
+        return false;
+      }
       out.write(snippetBytes);
+      if (state.getWrapper().contentLengthWasSet()) {
+        // Content-Length already set, need to update it.
+        state.getWrapper().updateContentLength();
+      }
```

Comment:
let's move updating the content-length prior to injecting, since by injecting, we may trigger a flush (if buffer fills up), in which case it will be too late to update the content length